src/GHC/Util.hs

```diff
@@ -325,3 +330,33 @@ wToStr :: Outputable a => W a -> String
 wToStr (W e) = showPpr baseDynFlags e
 instance Outputable a => Eq (W a) where (==) a b = wToStr a == wToStr b
 instance Outputable a => Ord (W a) where compare = compare `on` wToStr
+
+-- GHC parse trees don't contain pragmas. We work around this with
```

Comment:
What is in charge of reading, e.g. `LANGUAGE` pragmas, and setting parser configuration (for things like `LambdaCase`)?

Reply:
It's done in a pre-pass by a function called [`parseDynamicFilePragma`](https://hackage.haskell.org/package/ghc-lib-parser-8.8.0.20190723/docs/DynFlags.html#v:parseDynamicFilePragma) (or rather `parseDynamicFlagsCmdLine` and its implementation details). See http://blog.shaynefletcher.org/2019/06/have-ghc-parsing-respect-dynamic-pragmas.html. In this case, the parse is already deemed successful so I think this simple code is rigorous enough for the job.